framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -1252,6 +1262,90 @@ MeshDiagnosticsGenerator::checkLocalJacobians(const std::unique_ptr<MeshBase> &
                  num_negative_side_qp_jacobians);
 }
 
+void
+MeshDiagnosticsGenerator::checkNonMatchingEdges(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*Algorithm Overview
+    1)Prechecks
+      a)This algorithn only works for 3D so check for that first
+    2)Loop
+      a)Loop through every element
+      b)For each element get the edges associated with it
+      c)For each edge check overlap with any edges of nearby elements
+      d)Have check to make sure the same pair of edges are not being tested twice for overlap
+    3)Overlap check
+      a)Shortest line that connects both lines is perpendicular to both lines
+      b)A good overview of the math for finding intersecting lines can be found
+    here->paulbourke.net/geometry/pointlineplane/
+  */
+  if (mesh->mesh_dimension() != 3)
+    mooseError("The edge intersection algorithm only works with 3D meshes");
+  if (!mesh->is_serial())
+    mooseError("Only serialized/replicated meshes are supported");
+  unsigned int num_intersecting_edges = 0;
+  std::vector<Node> checked_edges;
+  for (auto elem : mesh->active_element_ptr_range())
+  {
+    std::vector<std::unique_ptr<Elem>> elem_edges(elem->n_edges());
+    for (auto i : elem->edge_index_range())
+      elem_edges[i] = elem->build_edge_ptr(i);
+    for (auto other_elem : mesh->active_element_ptr_range())
+    {
+      // If they're the same element then there's no need to check for overlap
+      if (elem == other_elem)
+      {
+        continue;
+      }
+      // Get bounding boxes for both elements. If the bounding boxes don't intersect then no edges
+      // will either
+      auto boundingBox1 = elem->loose_bounding_box();
+      auto boundingBox2 = other_elem->loose_bounding_box();
+      if (!(boundingBox1.intersects(boundingBox2)))
+      {
+        continue;
+      }
+      std::vector<std::unique_ptr<Elem>> other_edges(other_elem->n_edges());
+      for (auto j : other_elem->edge_index_range())
+        other_edges[j] = other_elem->build_edge_ptr(j);
+      for (auto & edge : elem_edges)
+      {
+        for (auto & other_edge : other_edges)
+        {
+          const auto node_list1 = edge->get_nodes();
+          const auto node_list2 = other_edge->get_nodes();
+          auto n1 = *node_list1[0];
+          auto n2 = *node_list1[1];
+          auto n3 = *node_list2[0];
+          auto n4 = *node_list2[1];
+          // Check if the edges have already been added to our check_edges list
+          if (std::find(checked_edges.begin(), checked_edges.end(), n1) != checked_edges.end() &&
+              std::find(checked_edges.begin(), checked_edges.end(), n2) != checked_edges.end() &&
```

Comment:
So dividing by 2 at the end would work in some cases but not all. If I don't have that check there then it will count the same edge for every cell that it's a part of so if could be counted twice or possibly more than that. If you want I could just take out the check and have it print out all that info. Or I could also turn checked_edges into a set instead. If I recall correctly finding elements in a std::set is O(log n) so it would at least scale better than searching through a vector of nodes?

Reply:
checked_edges into a set is good

but actually it s not checked-edges ? it's more like overlapping_edges_nodes?
in which case complexity is not quadratic (unless the whole mesh is filled with overlaps, which we dont expect)

so the check can stay